engine/src/test/java/io/camunda/zeebe/engine/state/appliers/EventAppliersTest.java

```diff
@@ -136,4 +141,33 @@ void shouldGetLatestVersionWhenMultipleRegisteredEventAppliersWithDifferentInten
     // then
     Assertions.assertEquals(expectedVersion, actualVersion);
   }
+
+  @Test
+  void shouldRegisterApplierForAllIntents() {
+    // given
+    final var events =
+        Intent.INTENT_CLASSES.stream()
+            .flatMap(c -> Arrays.stream(c.getEnumConstants()))
+            // Heuristic to detect event intents which are generally in past or present tense
+            // instead of the imperative used for commands.
+            .filter(intent -> intent.name().endsWith("ED") || intent.name().endsWith("ING"))
+            // CheckpointIntent is not handled by the engine
+            .filter(intent -> !(intent instanceof CheckpointIntent))
```

Comment:
🔧 Rather than filtering it out explicitly, we could add a `isRelatedTo()` method to `Intent` that informs whether the intent is related to the engine or the stream processor (or something else if we'd need that in the future). Such a check could be easier to maintain than this one.

Reply:
I've left that out for now. This only applies to `CheckpointIntent`, I don't expect we add any new intents like soon and if we do, this test will fail and it will be obvious how to fix it.